### PR TITLE
Cancel inv cfs

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/9316

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/cfs_service.py [hunk 1]
```diff
@@ -426,6 +434,33 @@ def add_nsf_adjustment(cls, cfs_account: CfsAccountModel, inv_number: str, amoun
...
+        adjustment_url = f'{cfs_base}/cfs/parties/{cfs_account.cfs_party}/accs/{cfs_account.cfs_account}/sites/' \
+                         f'{cfs_account.cfs_site}/invs/{inv_number}/adjs/'
+        current_app.logger.debug('Adjustment URL %s', adjustment_url)
+
+        adjustment = dict(
+            comment='Invoice cancellation',
+            lines=[
+                {
+                    'line_number': '1',
+                    'adjustment_amount': str(amount),
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/cfs_service.py:452]: Adjustment amount has to be how much amount needs to be deducted from the invoice. So if invoice amount is 100, then adjustment_amount should be `-100`.

File: pay-api/src/pay_api/services/cfs_service.py [hunk 2]
```diff
@@ -426,6 +434,33 @@ def add_nsf_adjustment(cls, cfs_account: CfsAccountModel, inv_number: str, amoun
...
+                    'activity_name': 'BC Registries - Invoice cancellation'
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/cfs_service.py:453]: activity_name is CFS_ADJ_ACTIVITY_NAME from constants file

File: pay-api/src/pay_api/services/internal_pay_service.py [hunk 3]
```diff
@@ -109,6 +109,11 @@ def process_cfs_refund(self, invoice: InvoiceModel):
         if routing_slip := RoutingSlipModel.find_by_number(routing_slip_number):
             routing_slip.remaining_amount += decimal.Decimal(invoice.total)
             routing_slip.flush()
+            invoice.invoice_status_code = InvoiceStatus.REFUND_REQUESTED.value
+        else:
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/services/internal_pay_service.py:113]: If it's not new FAS routing slip, we don't support refund. So it should be raising error, correct ?

File: pay-api/src/pay_api/services/cfs_service.py (hunk 1 above)
Comment (saravanpa-aot) [reply pay-api/src/pay_api/services/cfs_service.py:452]: Done

File: pay-api/src/pay_api/services/internal_pay_service.py (hunk 3 above)
Comment (saravanpa-aot) [reply pay-api/src/pay_api/services/internal_pay_service.py]: Good catch.Done

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**